src/promptflow-tracing/tests/e2etests/simple_functions.py

```diff
@@ -38,20 +39,35 @@ def greetings(user_id):
 
 
 @trace
-async def dummy_llm(prompt: str, model: str):
+async def dummy_llm_async(prompt: str, model: str):
     await asyncio.sleep(0.5)
     return "dummy_output"
 
 
+@trace
+def dummy_llm(prompt: str, model: str):
+    sleep(0.5)
+    return "dummy_output"
+
+
 @trace
 async def dummy_llm_tasks_async(prompt: str, models: list):
     tasks = []
     for model in models:
-        tasks.append(asyncio.create_task(dummy_llm(prompt, model)))
+        tasks.append(asyncio.create_task(dummy_llm_async(prompt, model)))
     done, _ = await asyncio.wait(tasks, return_when=asyncio.ALL_COMPLETED)
     return [task.result() for task in done]
 
 
+@trace
+def dummy_llm_tasks_threadpool(prompt: str, models: list):
+    prompts = [prompt] * len(models)
+    with ThreadPoolExecutorWithContext(2, "dummy_llm", initializer=lambda x: x, initargs=(prompt,)) as executor:
```

Comment:
Maybe add assert about checking customer's `initializer` will take effect?

Reply:
Let's add it in another PR